src/app/components/champion/ChampionSelector.tsx

```diff
@@ -1,46 +1,66 @@
-    const [championsList, setChampionsList] = useState<Champion[]>([]);
+    const [championId, setChampionId] = useState("Aatrox")
+    const championListState = useFetchAPI<Data>({data: {}}, () => axios.get(getUrlChampionList()))
+    const championState = useFetchAPI<Data>(undefined, () => axios.get(getUrlChampion(championId)), [championId])
 
-    useEffect(() => {
-        fetchChampionList()
-    }, [])
-
-    function fetchChampionList() {
-        axios.get(getUrlChampionList(), {}).then(response => {
-            const payload = response.data as Data
-            const champs = Object.values(payload.data) as Champion[]
-            setChampionsList(champs)
-        })
+    function extractChampionList(data: Data) {
+        return Object.values(data.data) as Champion[]
     }
 
-    function loadChampion(championSelected:string) {
-        axios.get(getUrlChampion(championSelected), {}).then(response => {
-            const payload = response.data as Data
-            const champs = Object.values(payload.data) as Champion[]
-            const champion = champs[0]
-            setChampion(champion as Champion)
-        })
+    function extractChampion(data: Data) {
+        return Object.values(data.data)[0] as Champion
     }
 
     const handleTextViewChange = useCallback(
-        (event, champion) => {
-            if (champion) {
-                loadChampion(champion.id)
+        (event, championChange) => {
+            if (championChange) {
+                setChampionId(championChange.id)
             }
         },
         [],
     );
 
+    const useStyles = makeStyles(() => ({
+        avatarSearch: {
+            height: '12%',
+            width: '12%',
+            marginRight: '5%'
+        }
+    }));
+
+    const classes = useStyles();
+
     return (
-        <ChampionView champion={champion} callback={handleTextViewChange} championsList={championsList}/>
+        <List component="nav" aria-label="secondary mailbox folders">
+            <Autocomplete
+                id="champion-box-complete"
+                options={extractChampionList(championListState.data as Data)}
+                getOptionLabel={(option) => option.name}
+                onChange={handleTextViewChange}
+                renderOption={(option) => (
+                    <React.Fragment>
+                        <Avatar className={classes.avatarSearch} variant='square'
+                                src={getUrlChampionAvatar(option.image.full)}/>
+                        {option.name}
+                    </React.Fragment>
+                )}
+                getOptionSelected={(option, value) => value.name === option.name}
+                renderInput={
+                    (params) => <TextField {...params} label="Champion" variant="outlined"/>
+                }
+            />
+            {championState.isLoading || !championState.data ? <h1>Loading your champion...</h1> :
```

Comment:
Loading your champion doesnt seem the best message. Spinner or nothing seems even better

Reply:
It is true that when actually selecting a champion, the text isn't really great looking. Will remove it for now, and eventually add a spinner later on.

Reply:
[Link to Trello ticket](https://trello.com/c/59VZ3beV/40-add-a-spinner-to-championselector)

Reply:
Ok for me